src/api/nlopt-in.hpp

```diff
@@ -89,10 +98,29 @@ namespace nlopt {
     typedef struct {
       opt *o;
       mfunc mf; func f; void *f_data;
+      functor_type functor;
```

Comment:
It seems like we'll need to make sure that this is initialized in the other places that call `new myfunc_data`, since otherwise `delete` on the data will call the destructor of an uninitialized value, no?

Reply:
This is actually OK because `std::function` has a default constructor (which simply stores a null pointer), so even when `new myfunc_data` is called, it is properly constructed; and then it's safe to delete it.